pkg/virt-controller/watch/vm_initializer.go

```diff
@@ -0,0 +1,353 @@
+				errors = append(errors, fmt.Errorf("spec.clock.timer: %v != %v", presetSpec.Clock.Timer, vmSpec.Clock.Timer))
+			}
+		}
+	}
+	if presetSpec.Features != nil && vmSpec.Features != nil {
+		if !reflect.DeepEqual(presetSpec.Features, vmSpec.Features) {
+			errors = append(errors, fmt.Errorf("spec.features: %v != %v", presetSpec.Features, vmSpec.Features))
+		}
+	}
+	if presetSpec.Devices.Watchdog != nil && vmSpec.Devices.Watchdog != nil {
+		if !reflect.DeepEqual(presetSpec.Devices.Watchdog, vmSpec.Devices.Watchdog) {
+			errors = append(errors, fmt.Errorf("spec.devices.watchdog: %v != %v", presetSpec.Devices.Watchdog, vmSpec.Devices.Watchdog))
+		}
+	}
+
+	if len(errors) > 0 {
+		return utilerrors.NewAggregate(errors)
+	}
+	return nil
+}
+
+func mergeDomainSpec(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.DomainSpec) (bool, error) {
+	presetConflicts := checkPresetMergeConflicts(presetSpec, vmSpec)
+	applied := false
+
+	if len(presetSpec.Resources.Requests) > 0 {
+		if vmSpec.Resources.Requests == nil {
+			vmSpec.Resources.Requests = k8sv1.ResourceList{}
+			for key, val := range presetSpec.Resources.Requests {
+				vmSpec.Resources.Requests[key] = val
+			}
+			applied = true
+		}
+	}
+	if presetSpec.CPU != nil {
+		if vmSpec.CPU == nil {
+			vmSpec.CPU = &kubev1.CPU{}
+			presetSpec.CPU.DeepCopyInto(vmSpec.CPU)
+			applied = true
+		}
+	}
+	if presetSpec.Firmware != nil {
+		if vmSpec.Firmware == nil {
+			vmSpec.Firmware = &kubev1.Firmware{}
+			presetSpec.Firmware.DeepCopyInto(vmSpec.Firmware)
+			applied = true
+		}
+	}
+	if presetSpec.Clock != nil {
+		if vmSpec.Clock == nil {
+			vmSpec.Clock = &kubev1.Clock{}
+			vmSpec.Clock.ClockOffset = presetSpec.Clock.ClockOffset
+			applied = true
+		}
+
+		if presetSpec.Clock.Timer != nil {
+			if vmSpec.Clock.Timer == nil {
+				vmSpec.Clock.Timer = &kubev1.Timer{}
+				presetSpec.Clock.Timer.DeepCopyInto(vmSpec.Clock.Timer)
+				applied = true
+			}
+		}
+	}
+	if presetSpec.Features != nil {
+		if vmSpec.Features == nil {
+			vmSpec.Features = &kubev1.Features{}
+			presetSpec.Features.DeepCopyInto(vmSpec.Features)
+			applied = true
+		}
+	}
+	if presetSpec.Devices.Watchdog != nil {
+		if vmSpec.Devices.Watchdog == nil {
+			vmSpec.Devices.Watchdog = &kubev1.Watchdog{}
+			presetSpec.Devices.Watchdog.DeepCopyInto(vmSpec.Devices.Watchdog)
+			applied = true
+		}
+	}
+	if presetConflicts != nil {
+		return applied, presetConflicts
+	}
+	return applied, nil
+}
+
+func applyPresets(vm *kubev1.VirtualMachine, presets []kubev1.VirtualMachinePreset, recorder record.EventRecorder) {
+	logger := log.Log
+	for _, preset := range presets {
+		applied, err := mergeDomainSpec(preset.Spec.Domain, &vm.Spec.Domain)
+		if err != nil {
+			msg := fmt.Sprintf("Unable to apply VirtualMachinePreset '%s': %v", preset.Name, err)
+			if applied {
+				msg = fmt.Sprintf("Conflicts occurred with VirtualMachinePreset '%s': %v",  preset.Name, err)
```

Comment:
There are two spaces between the ',' and preset.Name.  Whatever editor is in use should auto formatting things like this when the file is saved. When the editor doesn't perform the autoformatting, that means the next time someone opens and saves this file they'll get unexpected changes to their diff related to formatting. 

There are similar issues with vm_initializer_test.go and potentially some other files edited in this patch series if formatting isn't automatically enforced.

Reply:
actually, it looks like 'make' will do the autoformatting. Just run make and commit the changes.